src/debugger/buildConfig.ts

```diff
@@ -0,0 +1,317 @@
+import { Version } from "../utilities/version";
+import { TestKind, TestLibrary } from "../TestExplorer/TestRunner";
+
+/**
+ * Creates `vscode.DebugConfiguration`s for different combinations of
+ * testing library, test kind and platform. Use the static `swiftTestingConfig`
+ * and `xcTestConfig` functions to create
+ */
+export class DebugConfigurationFactory {
+    public static swiftTestingConfig(
+        ctx: FolderContext,
+        fifoPipePath: string,
+        testKind: TestKind,
+        testList: string[],
+        expandEnvVariables = false
+    ): vscode.DebugConfiguration | null {
+        return new DebugConfigurationFactory(
+            ctx,
+            fifoPipePath,
+            testKind,
+            TestLibrary.swiftTesting,
+            testList,
+            expandEnvVariables
+        ).build();
+    }
+
+    public static xcTestConfig(
+        ctx: FolderContext,
+        testKind: TestKind,
+        testList: string[],
+        expandEnvVariables = false
+    ): vscode.DebugConfiguration | null {
+        return new DebugConfigurationFactory(
+            ctx,
+            "",
+            testKind,
+            TestLibrary.xctest,
+            testList,
+            expandEnvVariables
+        ).build();
+    }
+
+    private constructor(
+        private ctx: FolderContext,
+        private fifoPipePath: string,
+        private testKind: TestKind,
+        private testLibrary: TestLibrary,
+        private testList: string[],
+        private expandEnvVariables = false
+    ) {}
+
+    /**
+     * Builds a `vscode.DebugConfiguration` for running tests based on four main criteria:
+     *
+     * - Platform
+     * - Toolchain
+     * - Test Kind (coverage, debugging)
+     * - Test Library (XCTest, swift-testing)
+     */
+    private build(): vscode.DebugConfiguration | null {
+        if (!this.hasTestTarget) {
+            return null;
+        }
+
+        switch (process.platform) {
+            case "win32":
+                return this.buildWindowsConfig();
+            default:
+                return this.buildDarwinConfg();
+        }
+    }
+
+    /* eslint-disable no-case-declarations */
+    private buildDarwinConfg(): vscode.DebugConfiguration | null {
+        switch (this.testLibrary) {
+            case "swift-testing":
+                let args = [
+                    "test",
+                    ...(this.testKind === TestKind.coverage ? ["--enable-code-coverage"] : []),
+                    "--enable-experimental-swift-testing",
+                    "--experimental-event-stream-version",
+                    "0",
+                    "--experimental-event-stream-output",
+                    this.fifoPipePath,
+                ];
+
+                if (this.swiftVersionGreaterOrEqual(6, 0, 0)) {
+                    args = [...args, "--disable-xctest"];
+                }
+
+                return {
+                    ...this.baseConfig,
+                    program: this.swiftProgramPath,
```

Comment:
You can't use `swift` executable if you are debugging tests. You have to call xctest or swift-testing executable directly.